### DISCUSSION TITLE
Icons are redrawn over one another in the status bar

### DISCUSSION THREAD

**Discussion by blaine12100** [lah7/gtk3-classic#77] [Help & Support]
Hi

So currently my WiFi icons are getting re drawn over one another. I checked the [Network Manger Wiki](https://wiki.archlinux.org/title/NetworkManager#nm-applet_tray_icons_display_wrongly) and they suggested to install the gtk3-classic package over gtk3 (I am currently using Arco Linux which is a Fork of Arch and using Qtile as my WM). I also installed the 32 bit version of this package also and I rebooted the machine. But the problem still persists.

Is there anything else that I can do from my end to solve this issue?

![ArcoLinux-2022-06-26-1656193922_screenshot_1366x768](https://user-images.githubusercontent.com/18356070/175791549-f90d49d2-4d5a-4eb1-aa01-3dfb5b998662.jpg)

**Comment by lah7** [comment]
Hi. I'm not quite sure what the bug is, or what desktop environment that is (MATE? XFCE?).

That line in the Arch Wiki seems outdated, since the patch in #12 seems to have gone, as it was apparently fixed upstream.

Regarding tray icons, I know there can be two technologies used to render tray icons: AppIndicator (or [Ayatana Indicators](https://ayatanaindicators.github.io/)) and GTK Status Icon (which is deprecated and buggy in my experience). Depending on the desktop environment used, you could swap the technology.

**Reply by blaine12100** [reply]
So I'm using Qtile Window Manager.

**Comment by lah7** [comment]
If it doesn't happen under normal `gtk3`, find out which is the problematic patch.

There is a script in this repository at [scripts/test-exclude-each-patch.sh](https://github.com/lah7/gtk3-classic/blob/master/scripts/test-exclude-each-patch.sh) which can help automate this testing -- it'll build everything individually excluding each patch. Then, it'll prompt to install each one. After installing, restart your desktop environment and see if that's the one. Alternately, you can manually comment (`#`) patches out in the `series` file and run `makepkg` as usual.

Let us know which excluded patch works, and we'll be able to figure what to do next.
